MOONCLOCK/apps.py

```diff
@@ -235,26 +247,85 @@ def update(self, first, remaining_duration):
         print('Recommended fees:', str(fees))
-
-        fee_string = str_cjust('{}{}:{}{}'.format(font.CHAR_SATOSHI, fastest_fee, font.CHAR_SATOSHI, hour_fee), 7)
+        
         if len(fastest_fee) + len(hour_fee) > 5:
-            self.display_group.clear()
-            self.display_group.render_string(fee_string, center=False)
+            fee_string = '{}{}:{}{}'.format(font.CHAR_SATOSHI, fastest_fee, font.CHAR_SATOSHI, hour_fee)
+            str_to_render = str_align(fee_string, 10, ' ', self.align)
         else:
-            self.display_group.clear()
-            self.display_group.render_string('{0} {1}'.format(font.CHAR_MONEY_BAG, fee_string), center=False)
-        self.display_group.show()  
+            fee_string = '{}{}:{}{}'.format(font.CHAR_SATOSHI, fastest_fee, font.CHAR_SATOSHI, hour_fee)
+            str_to_render = '{} {}'.format(font.CHAR_MONEY_BAG, str_align(fee_string, 7, ' ', self.align))
+        
+        self.display_group.clear()
+        self.display_group.render_string(str_to_render, center=False)
+        self.display_group.show()
 
 
 class Text(App):
-    def __init__(self, *args, text='', **kwargs):
+    def __init__(self, *args, text='', align='center', **kwargs):
         super().__init__(*args, **kwargs)
         self.text = text
+        self.align = align
 
     def update(self, first, remaining_duration):
         text = str(self.text)
         print('Show text: ', text)
 
         self.display_group.clear()
-        self.display_group.render_string(text, center=True)
-        self.display_group.show()  
+        self.display_group.render_string(str_align(text, 10, ' ', self.align), center=False)
+        self.display_group.show()
+
+class MarketCap(App):
+    CRYPTO_CHARACTER_MAP = {
+        'cardano': font.CHAR_ADA,
+        'baked-token': font.CHAR_BAKED,
+        'bitcoin': font.CHAR_BTC,
+        'dogecoin': font.CHAR_DOGE,
+        'ethereum': font.CHAR_ETH,
+        'litecoin': font.CHAR_LTC,
+        'polkadot': font.CHAR_POLKADOT,
+        'kusama': font.CHAR_KSM,
+    }
+
+    BASE_CURRENCY_CHARACTER_MAP = {
+        'usd': '$',
+        'eur': '€',
+        'gbp': '£',  
+        'sats': font.CHAR_WIDESATOSHI,
+        'btc': font.CHAR_BTC,
+    }
+
+    def __init__(self, *args, crypto='bitcoin', base_currency='usd', align='center', **kwargs):
+        super().__init__(*args, **kwargs)
+        self.crypto = crypto
+        self.base_currency = base_currency
+        self.align = align
+
+    def update(self, first, remaining_duration):
+        URL = 'https://api.coingecko.com/api/v3/coins/{}?localization=false&tickers=false&market_data=true&community_data=false&developer_data=false&sparkline=false'.format(self.crypto)
+
+        market_cap = self.requests.get(URL).json()['market_data']['market_cap'][self.base_currency]
+        
+        index=0
+        arr = ['', 'k', 'M', 'B', 'T', 'Q']
+        while market_cap >=1000:
+            index=index+1
+            market_cap /= 1000
```

Comment:
Code formatting... add spaces around `=`, `>=`, `+`, ...

Reply:
We should also export this function to a separate function, because some other applications might want to use it. Export it to the `utils` module, please.